test/spec/Feature/Query/CustomMediaSpec.hs

```diff
@@ -0,0 +1,176 @@
+module Feature.Query.CustomMediaSpec where
+
+import Network.Wai (Application)
+
+import Network.HTTP.Types
+import Network.Wai.Test    (SResponse (simpleBody, simpleHeaders))
+import Test.Hspec
+import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
+import Text.Heredoc        (str)
+
+import Protolude  hiding (get)
+import SpecHelper
+
+spec :: SpecWith ((), Application)
+spec = describe "custom media types" $ do
+  context "for tables with aggregate" $ do
+    it "can query if there's an aggregate defined for the table" $ do
+      r <- request methodGet "/lines" (acceptHdrs "application/vnd.twkb") ""
+      liftIO $ do
+        simpleBody r `shouldBe` readFixtureFile "lines.twkb"
+        simpleHeaders r `shouldContain` [("Content-Type", "application/vnd.twkb")]
+
+    it "can query by id if there's an aggregate defined for the table" $ do
+      r <- request methodGet "/lines?id=eq.1" (acceptHdrs "application/vnd.twkb") ""
+      liftIO $ do
+        simpleBody r `shouldBe` readFixtureFile "1.twkb"
+        simpleHeaders r `shouldContain` [("Content-Type", "application/vnd.twkb")]
+
+    it "will fail if there's no aggregate defined for the table" $ do
+      request methodGet "/lines" (acceptHdrs "text/plain") ""
+        `shouldRespondWith`
+        [json| {"code":"PGRST107","details":null,"hint":null,"message":"None of these media types are available: text/plain"} |]
+        { matchStatus  = 415
+        , matchHeaders = [matchContentTypeJson]
+        }
+
+    it "can get raw xml output with Accept: text/xml if there's an aggregate defined" $ do
+      request methodGet "/xmltest" (acceptHdrs "text/xml") ""
+        `shouldRespondWith`
+        "<myxml>foo</myxml>bar<foobar><baz/></foobar>"
+        { matchStatus = 200
+        , matchHeaders = ["Content-Type" <:> "text/xml; charset=utf-8"]
+        }
+
+  context "for tables with anyelement aggregate" $ do
+    it "will use the application/vnd.geo2+json media type for any table" $
+      request methodGet "/lines" (acceptHdrs "application/vnd.geo2+json") ""
+        `shouldRespondWith`
+        "\SOH{\"type\": \"FeatureCollection\", \"hello\": \"world\"}"
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/vnd.geo2+json"]
+        }
+
+    it "will use the more specific application/vnd.geo2 handler for this table" $ do
+      request methodGet "/shop_bles" (acceptHdrs "application/vnd.geo2+json") ""
+        `shouldRespondWith`
+        "\SOH\"anyelement overridden\""
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/vnd.geo2+json"]
+        }
+
+      request methodGet "/rpc/get_shop_bles" (acceptHdrs "application/vnd.geo2+json") ""
+        `shouldRespondWith`
+        "\SOH\"anyelement overridden\""
+        { matchStatus  = 200
+        , matchHeaders = ["Content-Type" <:> "application/vnd.geo2+json"]
+        }
```

Comment:
For some reason, these tests are giving a `\SOH` ([Start of Heading](https://www.ascii-code.com/character/%E2%90%81)) output. I've no idea if this comes from pg or Haskell, I think it's the latter.